test/dotcom/alerts/subway_test.exs

```diff
@@ -0,0 +1,229 @@
+defmodule Dotcom.Alerts.SubwayTest do
+  use ExUnit.Case
+
+  import Dotcom.Alerts.Subway
+  import Mox
+
+  import Test.Support.Generators.DateTime,
+    only: [random_date_time: 0, random_time_range_date_time: 1]
+
+  alias Test.Support.Factories
+
+  setup :verify_on_exit!
+
+  setup do
+    stub(Dotcom.Utils.DateTime.Mock, :coerce_ambiguous_date_time, fn date_time ->
+      Dotcom.Utils.DateTime.coerce_ambiguous_date_time(date_time)
+    end)
+
+    stub(Dotcom.Utils.DateTime.Mock, :now, fn ->
+      Dotcom.Utils.DateTime.now()
+    end)
+
+    :ok
+  end
+
+  describe "group_alerts/1" do
+    test "all groups are present" do
+      # Setup
+      group_order = group_order()
+      alerts = []
+
+      # Exercise
+      grouped_alerts = group_alerts(alerts)
+
+      # Verify
+      Enum.each(group_order, fn group ->
+        assert Map.has_key?(grouped_alerts, group)
+      end)
+    end
+
+    test "alerts default to an empty list" do
+      # Setup
+      alerts = []
+
+      # Exercise
+      group_counts = group_alerts(alerts)
+
+      # Verify
+      assert Enum.all?(group_counts, fn {_group, alerts} -> Enum.empty?(alerts) end)
+    end
+
+    test "alerts get grouped by effect" do
+      # Setup
+      groups = groups()
+      random_group = Enum.random(groups) |> Kernel.elem(0)
+
+      effects =
+        Enum.find(groups, fn {group, _effects} -> group == random_group end) |> Kernel.elem(1)
+
+      random_effect = if Enum.empty?(effects), do: :foobarbaz, else: Enum.random(effects)
```

Comment:
Same as below - `:foobarbaz` probably wants to be replaced with a fakerized value.

Also this is actually two different tests - one that validates that specified effects get grouped into specific buckets, and a different one that validates that other effects get grouped into `"Other"`. You can see that by the fact that the test itself has an if statement in it, along with a check that closely matches the (current) implementation.

Reply:
To wit, `:foobarbaz` shouldn't be fakerized because we need it to be an atom we know is not in the list of effects. That is...it *isn't* random. If Faker gave us something in the list of effects, the test would fail. It would be flaky.